src/main/java/org/verdictdb/coordinator/VerdictResultStream.java

```diff
@@ -16,47 +16,25 @@
 
 package org.verdictdb.coordinator;
 
-import org.verdictdb.connection.DbmsQueryResult;
-import org.verdictdb.core.resulthandler.ExecutionResultReader;
-
 import java.util.Iterator;
 
-public class VerdictResultStream
+public abstract class VerdictResultStream
```

Comment:
Would there be much change needed if we make this class an interface? Otherwise, I feel like interface may make more sense. What do you think?

Reply:
I think for now there is no big difference between abstract class and interface.

Reply:
Wait. I find VerdictResultStream also implement other interfaces.  And there are some override functions in VerdictResultStream. Is it good to move these functions to its subclass?

Reply:
No. An interface can also `extends` other interfaces.